src/qt/optionsmodel.cpp

```diff
@@ -404,7 +406,7 @@ bool OptionsModel::setData(const QModelIndex & index, const QVariant & value, in
             Q_EMIT coinControlFeaturesChanged(fCoinControlFeatures);
             break;
         case DatabaseCache:
-            if (settings.value("nDatabaseCache") != value) {
+            if (settings.value("nDatabaseCache") != value && QVariant::fromValue(gArgs.GetArg("-dbcache", nDefaultDbCache)) != value && value != QVariant::fromValue(nDefaultDbCache)) {
```

Comment:
What if someone has `dbcache=2000` in `bitcoin.conf` and then tries to reduce the dbcache via GUI back to 450?

Reply:
Currently that wouldn't do anything because we don't write to QSettings if the user sets the GUI dbcache value to the default. Should we switch such that we write to QSettings no matter what value the user sets?